cmd/internal/find/primary/meta/arrayPredicate.go

```diff
@@ -64,7 +64,7 @@ func parseArrayP(tokens []string, baseCaseParser, keySequenceParser predicate.Pa
 		}
 		return nil, nil, err
 	}
-	return arrayP(ptype, p), tokens, nil
+	return arrayP(ptype, p.(Predicate)), tokens, nil
```

Comment:
Did this commit really clean anything up? It seems to have resulted in moving a bunch of casts around.

Reply:
I guess it enforces the fact that all the parsers return the wrapped `Predicate` instead of `predicate.Predicate`, which might be good from a consistency standpoint. But may also be worth reverting it.

Reply:
I'll leave that up to you.